lib/config_cat/config_entry.ex

```diff
@@ -0,0 +1,85 @@
+defmodule ConfigCat.ConfigEntry do
+  @moduledoc false
+
+  alias ConfigCat.Config
+
+  defstruct config: %{}, etag: "", fetch_time_ms: 0, raw_config: "{}"
+
+  @type t :: %__MODULE__{
+          config: Config.t(),
+          etag: String.t(),
+          fetch_time_ms: non_neg_integer(),
+          raw_config: String.t()
+        }
+
+  @spec new(Config.t(), String.t(), String.t()) :: t()
+  def new(config, etag, raw_config) do
+    %__MODULE__{
+      config: config,
+      etag: etag,
+      fetch_time_ms: now(),
+      raw_config: raw_config
+    }
+  end
+
+  @spec new(Config.t(), String.t()) :: t()
+  def new(config, etag) do
+    %__MODULE__{
+      config: config,
+      etag: etag,
+      fetch_time_ms: now(),
+      raw_config: Jason.encode!(config)
+    }
+  end
+
+  @spec now() :: non_neg_integer()
+  def now, do: DateTime.utc_now() |> DateTime.to_unix(:millisecond)
+
+  @spec deserialize(String.t()) :: {:ok, t()} | {:error, String.t()}
+  def deserialize(str) do
+    with {:ok, [fetch_time_str, etag, raw_config]} <- parse(str),
+         {:ok, fetch_time_ms} <- parse_fetch_time(fetch_time_str),
+         :ok <- validate_etag(etag),
+         {:ok, config} <- parse_config(raw_config) do
+      {:ok,
+       %__MODULE__{
+         config: config,
+         etag: etag,
+         fetch_time_ms: fetch_time_ms,
+         raw_config: raw_config
+       }}
+    end
+  end
+
+  defp parse(str) do
+    case String.split(str, "\n", parts: 3) do
```

Comment:
`String.split(str, "\n", parts: 3)` is problematic because the config JSON may contain `\n` characters. An idea to handle this can be read the fetch_time and the etag first and read the rest of the data as config JSON. Please find a solution for this in the python SDK: https://github.com/configcat/python-sdk/blob/master/configcatclient/configentry.py#L25

Reply:
@kp-cat I believe that the `parts: 3` option addresses your concerns. It will return a list of at most 3 elements, where the first part is the fetch time, the second is the etag, and the third is whatever is left of the original string. That should be exactly equivalent to what the Ruby and Python versions are doing manually (and passes the same/similar tests as those versions).

Example for clarity: 
```elixir
iex(3)> String.split("a\nb\nc\nd\ne", "\n", parts: 3)
["a", "b", "c\nd\ne"]
```